lib/app/features/gallery/views/components/camera/camera_cell.dart

```diff
@@ -1,38 +1,104 @@
 // SPDX-License-Identifier: ice License 1.0
 
+import 'package:camera/camera.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:ice/app/extensions/extensions.dart';
+import 'package:ice/app/features/core/permissions/data/models/permissions_types.dart';
+import 'package:ice/app/features/core/permissions/providers/permissions_provider.dart';
+import 'package:ice/app/features/core/permissions/views/components/permission_aware_widget.dart';
+import 'package:ice/app/features/core/permissions/views/components/permission_dialogs/permission_sheets.dart';
 import 'package:ice/app/features/gallery/providers/providers.dart';
 import 'package:ice/app/features/gallery/views/components/camera/camera.dart';
-import 'package:ice/app/features/gallery/views/components/shimmer_loading_cell.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
 
-class CameraCell extends ConsumerWidget {
+class CameraCell extends HookConsumerWidget {
   const CameraCell({super.key});
 
   static double get cellHeight => 120.0.s;
   static double get cellWidth => 122.0.s;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final cameraControllerAsync = ref.watch(cameraControllerProvider);
-
-    return cameraControllerAsync.maybeWhen(
-      data: (controller) {
-        final isInitialized = controller?.value.isInitialized ?? false;
-
-        return GestureDetector(
-          onTap: () async => ref.read(galleryNotifierProvider.notifier).captureImage(),
-          child: SizedBox(
-            width: cellWidth,
-            height: cellHeight,
-            child: isInitialized && controller != null
-                ? CameraPreviewWidget(controller: controller)
-                : const CameraPlaceholderWidget(),
+    final hasPermission = ref.watch(hasPermissionProvider(Permission.camera));
+    final cameraControllerNotifier = ref.read(cameraControllerNotifierProvider.notifier);
+    final shouldOpenCamera = useState(false);
+
+    useOnAppLifecycleStateChange((_, AppLifecycleState current) async {
```

Comment:
Wdyt about moving this logic to the `CameraControllerNotifier`?

We can create an `AppLifecicleStateProvider` and use it inside other riverpod providers.

Reply:
done